managed/devops/bin/yb_platform_util.py

```diff
@@ -0,0 +1,700 @@
+        if data:
+            converted_data = json.dumps(data).encode('utf-8')
+            response = urlopen(request, converted_data)
+        else:
+            response = urlopen(request)
+        return convert_unicode_json(json.load(response))
+    
+
+    def __get_universe_by_name(self, universe_name):
+        """
+        Get universe data by name of the universe.
+
+        :param universe_name: Universe name.
+        :return: None
+        """
+        universe_url = '{0}/api/v1/customers/{1}/universes'.format(self.base_url, self.customer_uuid)
+        data = self.__call_api(universe_url)
+        for universe in data:
+            if universe.get('name') == universe_name:
+                del universe['pricePerHour']
+                return universe
+    
+    
+    def __create_universe_config(self, universe_data):
+        """
+        Create the universe config data from the json file. 
+        Remove extra fields from universe details before saving to the file.
+
+        :param universe_data: Stored universe data.
+        :return: Configured universe json.
+        """
+        configure_json = {}
+        clusters = copy.deepcopy(universe_data['universeDetails']['clusters'])
+        user_az_selected = universe_data['universeDetails']['userAZSelected']
+
+        excluded_keys = [
+            'uuid',
+            'storageType',
+            'awsArnString',
+            'useHostname',
+            'preferredRegion',
+            'regions',
+            'index',
+            'placementInfo'
+        ]
+
+        clusters_list = self.__get_cluster_list(clusters, excluded_keys)
+        configure_json['clusters'] = copy.deepcopy(clusters_list)
+        configure_json['clusterOperation'] = 'CREATE'
+        configure_json['userAZSelected'] = copy.deepcopy(user_az_selected)
+        configure_json['currentClusterType'] = 'PRIMARY'
+
+        return configure_json
+    
+
+    @staticmethod
+    def __get_cluster_list(clusters, excluded_keys):
+        """
+        Method to modify payload for create API. Modify user Intent inside cluster list.
+        Remove extra fields from clusters_list before saving to the file.
+
+        :param clusters: List of clusters.
+        :param excluded_keys: Keys to be excluded
+        :return: Cluster list.
+        """
+        clusters_list = []
+        for each_cluster in clusters:
+            cluster = {}
+            for key, val in each_cluster.items():
+                if key not in excluded_keys:
+                    user_intent = {}
+                    if key == 'userIntent':
+                        for key1, val1 in val.items():
+                            if key1 not in excluded_keys:
+                                # Remove unnecessary deviceinfo from create API payload.
+                                if key1 == 'deviceInfo':
+                                    val1.pop('diskIops')
+                                    val1.pop('storageType')
```

Comment:
Maybe write a comment on why this is unnecessary? Not clear why storageType is not required.

Reply:
Added. the main reason is present in func doc.

Reply:
Still not sure. Why is it not required?